### ISSUE TITLE
Observability for learning rate

### ISSUE THREAD

**Issue by DrChainsaw** [deeplearning4j/deeplearning4j#5687]
It would be nice to be able to observe the current learning rate from a Model, e.g. in a TrainingListener. I managed to ctrl+space my way into something like `((ComputationGraph) model).getOutputLayer(0).conf().getLayer().getUpdaterByParam("").getLearningRate(iteration, epoch)` but I guess it is not the most law-of-one-demeter-friendly approach one can think of.

I guess it also could be possible to store away an instance of the schedule or updater (or maybe even wrap it in some logging decorator) but I prefer to have as few dependencies as possible between setting model architecture and hyperparameters and the actual business of training or infering (including what to observe during training). This approach would probably also require some extra serialization effort just for the logging part.

Anyways, thanks for awesome software! I really appreciate the work you guys are doing!

**Comment by raver119** [comment]
Maybe we should tweak ScoreIterationListener to show it?

**Comment by raver119** [comment]
cc @AlexDBlack

**Comment by AlexDBlack** [comment]
I'dd add some helper methods (we have setLearningRate methods on MLN/CG now).
As for ScoreIterationListener: hm... in principle you can have different LRs on different layers (and even params!). But most of the time it's the same for all layers.

Maybe a learning rate listener would work here? It would track LR, and report when it changes (max. every N iterations).
